compiler/rustc_resolve/src/late/diagnostics.rs

```diff
@@ -1810,18 +1810,23 @@ impl<'a: 'ast, 'ast> LateResolutionVisitor<'a, '_, 'ast> {
         false
     }
 
-    fn let_binding_suggestion(&self, err: &mut Diagnostic, ident_span: Span) -> bool {
+    fn let_binding_suggestion(&mut self, err: &mut Diagnostic, ident_span: Span) -> bool {
         // try to give a suggestion for this pattern: `name = 1`, which is common in other languages
         let mut added_suggestion = false;
-        if let Some(Expr { kind: ExprKind::Assign(lhs, _rhs, _), .. }) = self.diagnostic_metadata.in_assignment &&
-            let ast::ExprKind::Path(None, _) = lhs.kind {
+        if let Some(Expr { kind: ExprKind::Assign(lhs, rhs, _), .. }) =
+            self.diagnostic_metadata.in_assignment
+        {
+            let is_rhs_assign = matches!(rhs.kind, ExprKind::Assign(..));
```

Comment:
does this actually have any effect?

Reply:
Yes, it will avoid this:

```rust
help: you might have meant to introduce a new binding
   |
13 |     let x = y = y;
   |     +++
```

Reply:
that seems fine I think?

Reply:
It's fine in syntax, the `x` will be `()`.
But maybe not the intention of users, it's Ok if you think we could suggest it.

Reply:
Yea, it keeps the logic here simpler, so let's just do that for now.